File: FacebookAccount.php
<?php
const FACEBOOK_UID = ""; // Facebook User ID
const FACEBOOK_XS = ""; // Get Your Facebook XS from your cookies

class FacebookAccount { 
            
	private $url = "https://m.facebook.com/home.php?sk=h_chr";
	private $cuser;
	private $xs;
	private $birthday_url = "https://m.facebook.com/browse/birthdays/";
    
	public function __construct ($cuser, $xs) {
		$this->cuser = $cuser;
		$this->xs = $xs;
	}
	
	public function getBirthdays () {
		
		$ch = curl_init();  
		curl_setopt($ch, CURLOPT_URL, $this->birthday_url);  
		curl_setopt($ch, CURLOPT_HEADER, 0);  
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch, CURLOPT_COOKIE, 'c_user='.$this->cuser.'; xs='.$this->xs.'; path=/');  
		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);  
		curl_setopt($ch, CURLOPT_VERBOSE, true);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch, CURLOPT_USERAGENT,'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0');
		curl_setopt($ch, CURLOPT_FILETIME, 1);  
		curl_setopt($ch, CURLOPT_TIMEOUT, 0);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYHOST,  2);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);  
		curl_setopt($ch, CURLOPT_REFERER, $this->birthday_url);  
		$content = curl_exec( $ch );
		$response = curl_getinfo( $ch );
		curl_close ( $ch );
		return $content;
	}
	
	public function getUserId ($userProfile) {
		$ch = curl_init();  
		curl_setopt($ch, CURLOPT_URL, "https://m.facebook.com/".$userProfile);  
		curl_setopt($ch, CURLOPT_HEADER, 0);  
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch, CURLOPT_COOKIE, 'c_user='.$this->cuser.'; xs='.$this->xs.'; path=/');  
		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);  
		curl_setopt($ch, CURLOPT_VERBOSE, true);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch, CURLOPT_USERAGENT,'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0');
		curl_setopt($ch, CURLOPT_FILETIME, 1);  
		curl_setopt($ch, CURLOPT_TIMEOUT, 0);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYHOST,  2);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);  
		curl_setopt($ch, CURLOPT_REFERER, "https://m.facebook.com/".$userProfile);  
		
		
		$content = curl_exec( $ch );

		$response = curl_getinfo( $ch );
		curl_close ( $ch );
		
		
		if (preg_match("/(\<a href=\"\/messages\/thread\/)([0-9]{1,20})/is", $content, $matches)) {
			return $matches[2];	
		}
		
	}
	
	
	
	public function sendMessage ($userID, $message="Happy Birthday 😊") {
		$ch = curl_init();  
		curl_setopt($ch, CURLOPT_URL, "https://mbasic.facebook.com/messages/compose/?ids=".$userID."");  
		curl_setopt($ch, CURLOPT_HEADER, 0);  
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch, CURLOPT_COOKIE, 'c_user='.$this->cuser.'; xs='.$this->xs.'; path=/');  
		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);  
		curl_setopt($ch, CURLOPT_VERBOSE, true);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
		curl_setopt($ch,CURLOPT_USERAGENT,'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0');
		curl_setopt($ch, CURLOPT_FILETIME, 1);  
		curl_setopt($ch, CURLOPT_TIMEOUT, 0);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYHOST,  2);  
		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);  
		curl_setopt($ch, CURLOPT_REFERER, "https://m.facebook.com/messages/read/?fbid=".$userID."&_rdr");  
		$content = curl_exec( $ch );
		$response = curl_getinfo( $ch );
		curl_close ( $ch );
		
		$post_fields = "";
		
		if (preg_match('/\<form method=\"post\" action=\"\/messages\/send\/\?icm=1"(.*)\<\/form>/iU', $content, $formMatches) == 1) {
			
			$formHTML = $formMatches[0];
			unset($formMatches);
			
			if (preg_match_all('/\<input type=\"hidden\" name=\"(.*)\" value=\"(.*)\"/iU', $formHTML, $formMatches) > 0) {
				
				for ($fieldCounter = 0; $fieldCounter<count($formMatches[1]); $fieldCounter++) {
					 $post_fields .= $formMatches[1][$fieldCounter] . "=" . rawurlencode($formMatches[2][$fieldCounter]) . "&";
				}
				
				$post_fields .= "body=".rawurlencode($message); 
				$post_fields .= "&send=Send"; 
				
			}
			
			
			$ch = curl_init();  
			curl_setopt($ch, CURLOPT_URL, "https://m.facebook.com/messages/send/?icm=1&amp;refid=12");  
			curl_setopt($ch, CURLOPT_HEADER, 0);  
			curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
			curl_setopt($ch, CURLOPT_COOKIE, 'c_user='.$this->cuser.'; xs='.$this->xs.'; path=/');  
			curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);  
			curl_setopt($ch, CURLOPT_POST, 1);  
			curl_setopt($ch, CURLOPT_POSTFIELDS, $post_fields);  
			curl_setopt($ch, CURLOPT_VERBOSE, true);
			curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);  
			curl_setopt($ch, CURLOPT_USERAGENT,'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0');
			curl_setopt($ch, CURLOPT_FILETIME, 1);  
			curl_setopt($ch, CURLOPT_TIMEOUT, 0);  
			curl_setopt($ch, CURLOPT_SSL_VERIFYHOST,  2);  
			curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);  
			curl_setopt($ch, CURLOPT_REFERER, "https://m.facebook.com/messages/send/?icm=1&amp;refid=12");  
			$content = curl_exec( $ch );
			$response = curl_getinfo( $ch );
			curl_close ( $ch );
			
			if (strpos(strip_tags($content), $message . "Just now") != false) {
				return true;
			}
			
			
			return false;
		
		}
		
		
					
		
		
		
	}
	
	public function sendBirthdays () {
		
		$content = $this->getBirthdays();
		$regex = "/\<h3\>(\<a(.*)href=\")(.*)(\">)(.*)\<\/h3\>/iUs";
		
		
		
		if (preg_match_all($regex, $content, $matches)) {
			
			foreach ($matches[3] as $userProfile) {
				$this->sendMessage($this->getUserId($userProfile));
			}
		}

	}

}


	$myAccount = new FacebookAccount(FACEBOOK_UID, FACEBOOK_XS);
	$myAccount->sendBirthdays();

?>

File: FacebookAccount.py
#!/usr/bin/env python
# -*- coding: utf-8 -*- 

from bs4 import BeautifulSoup
import requests
import re 

FACEBOOK_UID = "" 	# Facebook User ID
FACEBOOK_XS  = "" 	# Get Your Facebook XS from your cookies

class FacebookAccount(object):
	"""Facebook Auto Birthday Sender
	   We all have a lot of Facebook friends and sometimes we do not have time to send 'Happy Birthday' wishes to them.
	   This script was made for that purpose. Auto-Pilot your happy-birthday-wishes to your Facebook friends
	"""
	MAIN_URL 		= "https://mbasic.facebook.com/home.php?sk=h_chr"
	BIRTHDAY_URL    = "https://mbasic.facebook.com/browse/birthdays/"
	DEFAULT_USER_AGENT 	= 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0'

	def __init__(self, fb_xs, fb_cuser):
		self.fb_xs = fb_xs
		self.fb_cuser = fb_cuser

	def getBirthdays(self):
		headers = { 'User-Agent' : self.DEFAULT_USER_AGENT, 'Referer' : self.BIRTHDAY_URL }
		cookies = { 'xs': self.fb_xs, 'c_user': self.fb_cuser}
		response = requests.get(self.BIRTHDAY_URL, cookies=cookies, headers=headers)
		return response.text

	def getUserID(self, userProfile):
		headers = { 'User-Agent' : self.DEFAULT_USER_AGENT, 'Referer' : "https://mbasic.facebook.com/" + str(userProfile) }
		cookies = { 'xs': self.fb_xs, 'c_user': self.fb_cuser}
		response = requests.get("https://mbasic.facebook.com/" + str(userProfile), cookies=cookies, headers=headers)
		matches = re.findall(r"\<a href=\"\/messages\/thread\/([0-9]{1,20})", response.text)
		if matches:
			return "".join(matches)
		else:
			return -1

	def sendMessage(self, userID, message="Happy Birthday 😊"):
		headers = { 'User-Agent' : self.DEFAULT_USER_AGENT, 'Referer' : 'https://m.facebook.com/messages/read/?fbid='+str(userID) }
		cookies = { 'xs': self.fb_xs, 'c_user': self.fb_cuser}
		firstRes = requests.get('https://mbasic.facebook.com/messages/compose/?ids='+str(userID), cookies=cookies, headers=headers)
		firstResParser = BeautifulSoup(firstRes.text, "html.parser")
		formObject = firstResParser.find("form", attrs={'id': 'composer_form'})
		allInputs = formObject.find_all('input', attrs={'type': 'hidden'})
		post_fields = {}
		for singleInput in allInputs:
			if singleInput:
				post_fields[singleInput.get('name')] = singleInput.get('value')
		post_fields['body'] = message
		post_fields['send'] = 'Send'
		finalRes = requests.post("https://mbasic.facebook.com/messages/send/?icm=1&amp;refid=12", data=post_fields, cookies=cookies, headers=headers)

		if finalRes.status_code == 200:
			return True
		else:
			return False

	def sendBirthdays(self):
		successFlag = 0
		content = self.getBirthdays()
		contentParser = BeautifulSoup(content, "html.parser")
		bdDiv = contentParser.find("div", attrs={'id': 'objects_container'})
		bdPeople = bdDiv.find_all("h3")
		for bdPerson in bdPeople:
			myFriendUsername = bdPerson.find("a").get('href')
			self.sendMessage(self.getUserID(myFriendUsername))
			successFlag += 1
		return successFlag


myAccount = FacebookAccount(FACEBOOK_XS, FACEBOOK_UID)
print(myAccount.sendBirthdays())
